libraries/chain/controller.cpp

```diff
@@ -4240,15 +4250,13 @@ struct controller_impl {
          if (bsp->is_recent() || testing_allow_voting) {
             if (use_thread_pool == use_thread_pool_t::yes && async_voting == async_t::yes) {
                boost::asio::post(thread_pool.get_executor(), [this, bsp=bsp]() {
-                  const auto& latest_qc_claim__block_ref = bsp->core.get_block_reference(bsp->core.latest_qc_claim().block_num);
-                  if (fork_db_validated_block_exists(latest_qc_claim__block_ref.block_id)) {
+                  if (fork_db_validated_block_exists(bsp->previous(), bsp->core.latest_qc_claim().block_num)) {
```

Comment:
I think you have to pass `latest_qc_claim__block_ref.block_id` instead of the block_num.

Reply:
I don't think so, as long as we check it is an ancestor of the passed `id`.